src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -184,16 +184,16 @@ void RaftReplDev::on_fetch_data_received(intrusive< sisl::GenericRpcData >& rpc_
     ctx->outstanding_read_cnt = fetch_req->request()->entries()->size();
```

Comment:
I don't think we need to do Context and read count etc, which is reimplementing future/promise right? Suggest to create promise and move promise to callback and do setValue there. We don't need mutex, cv and atomic count explicltly.

Reply:
We discussed this, somehow I was hitting hang while using the future/promise by collectAll on fetch side (receiving side is already calling collectAllUnsafe). Need to dig into why.